abad_test.go

```diff
@@ -46,18 +49,59 @@ func TestNumberEval(t *testing.T) {
 			code: "-+-+0",
 			obj:  types.Number(0.0),
 		},
+		{
+			code: "0.1.",
+			err:  E("<anonymous>:1:0: invalid token: 0.1."),
```

Comment:
I have serious doubts about testing error messages like this, like if a space is removed from the message the test fails =/. But just testing if there is an error may allow the test pass for a completely different reason/error. I'm starting to think on a more high level explicit error handling package like the one developed on upspin, perhaps with something like that we could have more higher level/non formatting validation like checking if the error is a "invalidToken" error (could be a function that checks the type of the errors, and there would be types of errors =P).

Not sure, this is pretty much new territory for me, trying to think about errors as first class design concerns.

Reply:
I have the feeling also, but in case of upspin the error had to travel from server to client (by net) and here the error is the string presented to the programmer.
I'm not saying we cannot create an error package, but when we create error types we get other challenges like how to compare errors.
I would like a proposal to solve this, but I think is minor in comparison with the scope of the project,

Reply:
the first idea is to do at least type comparison and have different types of errors (like lexical ones). but it is not a priority indeed =)